src/network/networkd-network.c

```diff
@@ -229,6 +229,19 @@ int network_verify(Network *network) {
         if (network->ignore_carrier_loss < 0)
                 network->ignore_carrier_loss = network->configure_without_carrier;
 
+        if (!IN_SET(network->activation_policy, ACTIVATION_POLICY_UP, ACTIVATION_POLICY_ALWAYS_UP)) {
+                if (network->required_for_online < 0) {
+                        log_debug("%s: ActivationPolicy= is not up and RequiredForOnline= is not set, "
+                                  "Setting RequiredForOnline=false.", network->filename);
```

Comment:
`"%s: setting RequiredForOnline=no because ActivationPolicy=%s", network->filename, activation_policy_to_string(network->activation_policy))"` ?

Reply:
thanks that is definitely much better, changed in latest push